### ISSUE TITLE
Unable to calculate charge curve automatically on boot

### ISSUE THREAD

**Issue by tomtaylor** [springfall2008/batpred#1480]
**Describe the bug**

Predbat seems to not be able to calculate the charge curve automatically on boot. I see the following log lines:

```
21971	2024-09-23 20:23:13.102041: Found 1 inverters totals: min reserve 0.38 current reserve 0.38 soc_max 9.52 soc 1.14 charge rate 3.5999999999999996 kW discharge rate 3.5999999999999996 kW battery_rate_min 0.0 w ac limit 3.6 export limit 3.6 kW loss charge 3 % loss discharge 3 % inverter loss 4 %
21970	2024-09-23 20:23:13.096372: Note: Cannot find battery charge curve (no final curve), one of the required settings for predbat_status, soc_kw, battery_power and charge_rate do not have history, check apps.yaml
21969	2024-09-23 20:23:13.074284: Find discharge curve has 5.0 days of data, max days 5
21968	2024-09-23 20:23:11.534461: Find discharge curve with sensors sensor.givtcp_fa2314g929_soc_kwh and number.givtcp_fa2314g929_battery_discharge_rate and predbat.status and sensor.givtcp_fa2314g929_battery_power
21967	2024-09-23 20:23:11.528220: Note: Cannot find battery charge curve (no final curve), one of the required settings for predbat_status, soc_kw, battery_power and charge_rate do not have history, check apps.yaml
21966	2024-09-23 20:23:11.505998: Find charge curve has 5.0 days of data, max days 5
21965	2024-09-23 20:23:09.700972: Find charge curve with sensors sensor.givtcp_fa2314g929_soc_kwh and number.givtcp_fa2314g929_battery_charge_rate and predbat.status and sensor.givtcp_fa2314g929_battery_power
```

However, all the entities are present and have enough data as far as I can tell.

**Expected behavior**

It outputs the predicted charge curve on boot.

**Predbat version**

8.4.9

**Environment details**

I'm using a Givenergy 9.5kWh gen 2 + hybrid inverter gen 3 3.6kW. Standard HAOS setup installed with Addon.

I appreciate there's not much to go on there. Let me know if I can provide more details somehow.

**Comment by gcoan** [comment]
In the 5 days of history have you fully charged and discharged the battery to totally full and empty?  

If you haven't then That's most likely why predbat can't determine the charge / discharge curves.

**Comment by tomtaylor** [comment]
@gcoan I thought I had, but let me force a charge to 100% and see what happens!

**Comment by gcoan** [comment]
Another thing to check, make sure that your apps.yaml has:

```
battery_power:
  - sensor.givtcp_{geserial}_battery_power
```

There was a bug fix in 8.4.0 to correct the entity name in apps.yaml for battery_power, pv_power and load_power - all of these previously were number.givtcp\_xxxx and they need to be sensor.givtcp\_xxxx

**Comment by tomtaylor** [comment]
This works now, thanks.
